### PR TITLE
Add A Guide on How to Handle Exceptions

### PR DESCRIPTION
**Pull request by brfrn169**
This PR adds A Guide on How to Handle Exceptions. Please take a look!

### CODE REVIEW

File: core/src/main/java/com/scalar/db/api/DistributedTransaction.java [hunk 1]
```diff
@@ -63,9 +65,11 @@
    *
    * @param get a {@code Get} command
    * @return an {@code Optional} with the returned result
+   * @throws CrudConflictException if transaction conflict happened. You can retry the transaction
```
Comment (brfrn169) [inline core/src/main/java/com/scalar/db/api/DistributedTransaction.java:68]: I also added javadoc and throws declarations for `XXXConflictException`.

**Review by feeblefakie — COMMENTED** [review]
Overall looking good! Thank you!
I left some comments and suggestions.

**Comment by brfrn169** [comment]
@feeblefakie I fixed it based on your review. Please take a look again when you have time!

**Comment by brfrn169** [comment]
As discussed, I changed the doc title to "A Guide on How to Handle Exceptions". Please take a look! @feeblefakie @thongdk8

**Review by feeblefakie — APPROVED** [review]
LGTM! Thank you!

**Review by thongdk8 — APPROVED** [review]
LGTM! Thank you!

**Pull request merged**